ereolen_support/ereolen_support.helper.inc

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * @file
+ * Contains helper functions for the support form.
+ */
+
+/**
+ * Search for a drupal user email in Jira.
+ *
+ * @param \stdClass $user
+ *   A drupal user.
+ *
+ * @return bool|mixed
+ *   A jira user or FALSE if no user was found.
+ *
+ * @throws \Exception
+ */
+function _ereolen_support_search_jira_user($user, $form_state) {
+  $form_email = $form_state['values']['ereolen_support_email'];
+  $requestParts = _ereolen_support_get_request_parts('GET');
+  $result = drupal_http_request($requestParts['jira_path'] . '/user/search?includeInactive=true&username=' . $form_email, $requestParts['options']);
+  $resultArray = json_decode($result->data);
```

Comment:
Check that it actually returns valid json before trying to parse it the `drupal_http_request` returns http error codes etc and could be used to give user feedback or at least log error with watchdog for debugging later on.

This is true for all `drupal_http_request` in this file.

Reply:
The watchdog will not be flooded as you simply filter the logs in UI or using grep. The watchdog logs should be named with the modules name as type.

Reply:
The issue with json validation is that `json_decode($result->data);` assumes that the server was there an responded with a body in `->data` and json_decode will return `FALSE` if it's not json. I known that you check that it's an array.

But after getting the result you call  `_ereolen_support_log_error` which could display an error message to the end-user but at the same time you return false from this function. An I think it's the responsibility for the caller to notify the user.

Reply:
Added _ereolen_support_result_confirmed($result) function to check for $result->status 200 or 201 on all jira calls. That function may need other checks to ensure proper error handling. User notifications are removed from here.